era5cli/key_management.py

```diff
@@ -0,0 +1,173 @@
+        InvalidRequestError: If the test request failed, likely due to changes in the
+            CDS API's variable naming.
+
+    Returns:
+        True if a connection was made succesfully.
+    """
+    connection = cdsapi.Client(
+        url=url,
+        key=fullkey,
+        verify=True,
+        quiet=True,  # Supress output to the console from the test retrieve.
+    )
+
+    try:
+        # Check the URL
+        connection.status()  # pragma: no cover
+
+        # Checks if the authentication works, without downloading data
+        connection.retrieve(  # pragma: no cover
+            "reanalysis-era5-single-levels",
+            {
+                "variable": "2t",
+                "product_type": "reanalysis",
+                "date": "2012-12-01",
+                "time": "14:00",
+                "format": "netcdf",
+            },
+        )
+        return True
+
+    except ConnectionError as err:
+        raise ConnectionError(
+            f"{os.linesep}Failed to connect to CDS. Please check your internet "
+            "connection and/or the"
+            f" URL in the era5cli configuration: {ERA5CLI_CONFIG_PATH.resolve()}"
+            f"{os.linesep}Or redefine your configuration with 'era5cli config'"
+        ) from err
+
+    except Exception as err:
+        if AUTH_ERR_MSG in str(err):
+            raise InvalidLoginError(
+                f"{os.linesep}Authorization with the CDS served failed. Likely due to"
+                " an incorrect key or UID."
+                f"{os.linesep}Please check your era5cli configuration file: "
+                f"{ERA5CLI_CONFIG_PATH.resolve()}{os.linesep}"
+                "Or redefine your configuration with 'era5cli config'"
+            ) from err
+        if NO_DATA_ERR_MSG in str(err):
+            raise InvalidRequestError(
+                f"{os.linesep}Something changed in the CDS API. Please raise an issue "
+                "on https://www.github.com/eWaterCycle/era5cli"
+            ) from err
+        raise err  # pragma: no cover
+
+
+def run_config(
+    url: str,
+    uid: str,
+    key: str,
+) -> True:
+    """Check the user-input configuration. Entry point for the CLI."""
+    try:
+        attempt_cds_login(url, fullkey=f"{uid}:{key}")
+        write_era5cli_config(url, uid, key)
+        print(
+            f"Keys succesfully validated and stored in {ERA5CLI_CONFIG_PATH.resolve()}"
+        )
+        return True
+    except InvalidLoginError:
+        print(
+            "Error: the UID and key are rejected by the CDS. "
+            "Please check and try again."
+        )
+    return False
+
+
+def check_era5cli_config() -> None:
+    """Validate if the era5cli config exists, and can connect to the CDS.
+
+    If no era5cli config file exists, but a CDS api file exists in the default location,
+    This routine will attempt to use those keys, and ask the user if they want to use
+    these.
```

Comment:
Wonder if this should be some kind of deprecation message

Reply:
I could modify the code so it raises a deprecation warning if a .cdsapirc file exists, but the era5cli config does not. However, this would lead to new era5cli users also getting this deprecation warning if they used the cdsapi on their machine.

As era5cli is command line only, I am not to worried about changing the user experience slightly, as we would not be breaking anyone's workflow/code. They will just see this migration message when they run era5cli the first time. This PR makes no modifications to the actual requests, so user's old commands still work in the same way.